doc/guides/west/manifest.rst

```diff
@@ -788,6 +787,118 @@ In this case, both ``groupA`` and ``groupB`` are disabled.
+For complete details, see :ref:`west-manifest-formal`.
+
+.. warning::
+
+   The below semantics apply to west v0.10.0 and later. West v0.9.x semantics
+   are different, and combining ``group-filter`` with ``import`` in west v0.9.x
+   is discouraged.
+
+In short:
+
+- if you only import one manifest, any groups it disables in its
+  ``group-filter`` are also disabled in your manifest
+- you can override this in your manifest file's ``manifest: group-filter:``
+  value, your workspace's ``manifest.group-filter`` configuration option, or
+  both
+
+Here are some examples.
+
+Example 1: no overrides
+-----------------------
+
+You are using this :file:`parent/west.yml` manifest:
+
+.. code-block:: yaml
+
+   # parent/west.yml:
+   manifest:
+     projects:
+       - name: child
+         url: https://git.example.com/child
+         import: true
+       - name: project-1
+         url: https://git.example.com/project-1
+         groups:
+           - unstable
+
+And :file:`child/west.yml` contains:
+
+.. code-block:: yaml
+
+   # child/west.yml:
+   manifest:
+     group-filter: [-unstable]
+     projects:
+       - name: project-2
+         url: https://git.example.com/project-2
+       - name: project-3
+         url: https://git.example.com/project-3
+         groups:
+           - unstable
+
+Only ``child`` and ``project-2`` are active in the resolved manifest.
+
+The ``unstable`` group is disabled in :file:`child/west.yml`, and that is not
+overridden in :file:`parent/west.yml`. Therefore, the final ``group-filter``
+for the resolved manifest is ``[-unstable]``.
+
+Since ``project-1`` and ``project-3`` are in the ``unstable`` group and are not
+in any other group, they are inactive.
+
+Example 2: overriding an imported ``group-filter``
+--------------------------------------------------
+
+You are using this :file:`parent/west.yml` manifest:
+
+.. code-block:: yaml
+
+   # parent/west.yml:
+   manifest:
+     group-filter: [+unstable,-optional]
+     projects:
+       - name: child
+         url: https://git.example.com/child
+         import: true
+       - name: project-1
+         url: https://git.example.com/project-1
+         groups:
+           - unstable
+
+And :file:`child/west.yml` contains:
+
+.. code-block:: yaml
+
+   # child/west.yml:
+   manifest:
+     group-filter: [-unstable]
+     projects:
+       - name: project-2
+         url: https://git.example.com/project-2
+         groups:
+           - optional
+       - name: project-3
+         url: https://git.example.com/project-3
+         groups:
+           - unstable
+
+Only the ``child``, ``project-1``, and ``project-3`` projects are active.
+
+The ``[-unstable]`` group filter in :file:`child/west.yml` is overridden in
+:file:`parent/west.yml`, so the ``unstable`` group is enabled. Since
+``project-1`` and ``project-3`` are in the ``unstable`` group, they are active.
+
+The same :file:`parent/west.yml` file disables the ``optional`` group, so
+``project-2`` is inactive.
+
```

Comment:
Great section! The only thing I am missing in it is perhaps a mention to how `west config manifest.group-filter` can modify this. You did mention this in the beginning of the section, but perhaps a third example that replicates the second one while also adding a call to `west config manifest.group-filter +optional` so that we show all projects enabled?

Reply:
>  perhaps a third example

Good idea. Now added.